src/jira.js

```diff
@@ -0,0 +1,1016 @@
+      method: 'POST',
+      followAllRedirects: true,
+      json: true,
+      body: version
+    };
+    return this.doRequest(requestOptions);
+  }
+
+  // ## Update a version ##
+  // ### Takes ###
+  // *  version: an object of the new version
+  // ### Returns ###
+  // *  version: should be the same version you passed up
+  // [Jira Doc](https://docs.atlassian.com/jira/REST/latest/#d2e510)
+  /* {
+   *    "id": The ID of the version being updated. Required.
+   *    "description": "An excellent version",
+   *    "name": "New Version 1",
+   *    "archived": false,
+   *    "released": true,
+   *    "releaseDate": "2010-07-05",
+   *    "userReleaseDate": "5/Jul/2010",
+   *    "project": "PXA"
+   * }
+   */
+  updateVersion(version) {
+    const requestOptions = {
+      rejectUnauthorized: this.strictSSL,
+      uri: this.makeUri(`/version/${version.id}`),
+      method: 'PUT',
+      followAllRedirects: true,
+      json: true,
+      body: version
+    };
+
+    return this.doRequest(requestOptions);
+  }
+
+  // ## Pass a search query to Jira ##
+  // ### Takes ###
+  // *  searchString: jira query string
+  // *  optional: object containing any of the following properties
+  //   *  startAt: optional index number (default 0)
+  //   *  maxResults: optional max results number (default 50)
+  //   *  fields: optional array of desired fields, defaults when null:
+  //     *  "summary"
+  //     *  "status"
+  //     *  "assignee"
+  //     *  "description"
+  // ### Returns ###
+  // *  issues: array of issues for the user
+  // [Jira Doc](https://docs.atlassian.com/jira/REST/latest/#d2e4424)
+  searchJira(searchString, optional) {
+    const theOptional = optional || {};
+    const requestOptions = {
+      rejectUnauthorized: this.strictSSL,
+      uri: this.makeUri('/search'),
+      method: 'POST',
+      json: true,
+      followAllRedirects: true,
+      body: {
+        jql: searchString,
+        ...theOptional
+      }
+    };
+
+    return this.doRequest(requestOptions);
+  }
+
+  // ## Search user on Jira ##
+  // ### Takes ###
+  // username: A query string used to search username, name or e-mail address
+  // startAt: The index of the first user to return (0-based)
+  // maxResults: The maximum number of users to return (defaults to 50).
+  // includeActive: If true, then active users are included in the results (default true)
+  // includeInactive: If true, then inactive users are included in the results (default false)
+  // ### Returns ###
+  // *  users: array of users for the user
+  // [Jira Doc](http://docs.atlassian.com/jira/REST/latest/#d2e3756)
+  searchUsers({username, startAt, maxResults, includeActive, includeInactive}) {
+    const requestOptions = {
+      rejectUnauthorized: this.strictSSL,
+      uri: this.makeUri(`/user/search?username=${username}&startAt=${startAt || 0}&maxResults=${maxResults || 50}&includeActive=${includeActive || true}&includeInactive=${includeInactive || false}`),
```

Comment:
This url can be cleaned up with the querystring serialization options. See: https://www.npmjs.com/package/request#request-options-callback